client/fingerprint_manager.go

```diff
@@ -86,25 +111,93 @@ func (fm *FingerprintManager) setupDrivers(drivers []string) error {
 
 		p, period := d.Periodic()
 		if p {
-			go fm.run(d, period, name)
+			go fm.runFingerprint(d, period, name)
+		}
+
```

Comment:
I think what this should do is not use `runFingerprint` or `runHealthCheck` but instead gather the fact that the Driver is health checking or periodic and pass all that to a new method that runs the drivers fingerprint/healthcheck periodically and then take the returned info and properly construct the DriverInfo object (merged health check and attributes)

Reply:
This is a bit tricky as the fingerprint/healthcheck methods run on different intervals, so if we merge them in lockstep then we only update the node as often has the slowest interval. Thinking through ways to do this more cleanly.